### PR TITLE
First version of a higher level arm API

### PR DESCRIPTION
**Pull request by SyrianSpock**
This PR introduces a higher level wrapper to arm trajectories `ArmTrajectory`.
It provides a fluent interface, a typical usage looks like

```c++
const auto trajectory = ArmTrajectory(&arm)
                            .startAt({1, 2, 3, COORDINATE_ARM})
                            .goThrough({4, 5, 6, COORDINATE_ROBOT})
                            .goThrough({7, 8, 9, COORDINATE_TABLE});
trajectory.execute();
wait_s(trajectory.duration());
```

And trajectories can be chained as follow
```c++
const auto trajectory = ArmTrajectory(&arm)
                            .startAt({1, 2, 3, COORDINATE_ARM})
                            .goThrough({4, 5, 6, COORDINATE_ROBOT})
                            .goThrough({7, 8, 9, COORDINATE_TABLE});
const auto finalFrame = trajectory.execute();

// do other stuff

const auto secondTrajectory = ArmTrajectory(&arm)
                                .startAt(finalFrame)
                                .goThrough({4, 5, 6, COORDINATE_ROBOT})
                                .goThrough({7, 8, 9, COORDINATE_TABLE});
```

### CODE REVIEW

File: master-firmware/src/arms/arm.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#ifndef ARM_API_H
```
Comment (antoinealb) [inline master-firmware/src/arms/arm.h]: Maybe we should name C++ headers files with ".hpp".

Comment (SyrianSpock) [reply master-firmware/src/arms/arm.h]: Not sure what we should have as standard

**Comment by SyrianSpock** [comment]
The idea is to try out this interface and see what are the pain points. This version is just C wrapped in a class to have a fluent interface (syntactic sugar).

**Comment by nuft** [comment]
> More C++ might be dangerous though, is this a tradeoff that we want ?

I don't see this as problematic, especially if we keep things simple.
In case of the strategy code it makes sense to use C++. It should be clean and easy to write.

**Pull request merged**